### PR TITLE
Add types

### LINKED ISSUE #3: Please, add typescript types definitions to this module
### Problem

I'm using this lib with TypeScript and the TS compiler complains about the lack of typings:

```
Could not find a declaration file for module 'remark-unlink'. 'xxxx/node_modules/remark-unlink/index.js' implicitly has an 'any' type.
  Try `npm install @types/remark-unlink` if it exists or add a new declaration (.d.ts) file containing `declare module 'remark-unlink';`
ts(7016)
```

### Solution

Create a `types/index.d.ts` file pretty much like [strip-markdown](https://github.com/remarkjs/strip-markdown/blob/625506cb8377c076663001273fde30f9ffae03ee/types/index.d.ts), and update the `files` field of your `package.json`.

I think that the following will be enough:

```typescript
declare module 'remark-unlink' {
  import type { Plugin } from 'unified';

  /**
   * Remark plugin to remove Markdown links, images, references, and definitions.
   */
  declare const remarkUnlink: Plugin;

  export = remarkUnlink;
}
```

### PR DESCRIPTION
**Pull request by micalevisk**
Closes #3 :smiling_face_with_three_hearts:

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// TypeScript Version: 3.8
+
+import type {Plugin} from 'unified'
+
+/**
+ * Remark plugin to remove Markdown links, images, references, and definitions.
+ */
+declare const remarkUnlink: Plugin
```
Comment (ChristianMurphy) [inline types/index.d.ts]: this plugin does not accept settings
```suggestion
declare const remarkUnlink: Plugin<[]>
```

**Review by ChristianMurphy — APPROVED** [review]
Thanks @micalevisk!

File: types/index.d.ts (hunk 1 above)
Comment (micalevisk) [reply types/index.d.ts]: right! ty I'll squash this change

Comment (micalevisk) [reply types/index.d.ts]: actually with this I'll get

```
  types/index.d.ts:8:36
  ✖  8:36  Don't use [] as a type. Don't use the empty array type []. It only allows empty arrays. Use SomeType[] instead.  @typescript-eslint/ban-types
```

Comment (ChristianMurphy) [reply types/index.d.ts]: From xo?
Yeah, xo is good at linting full TS files but not designed for dts files.
https://github.com/remcohaszing/mdast-util-definitions/blob/b8b1302c70e2387247b97d4469742608491a1237/package.json#L49
is an approach used elsewhere. DTSlint is better aligned with and designed for dts file checking.

**Review by ChristianMurphy — APPROVED** [review]

**Review by Murderlon — APPROVED** [review]

**Review by remcohaszing — APPROVED** [review]

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
Thanks all, released!

**Pull request merged**